From 1ca3da2468482e407b86339101e29a96f652e44e Mon Sep 17 00:00:00 2001
Message-ID: <1ca3da2468482e407b86339101e29a96f652e44e.1732384228.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732384228.git.sharlatanus@gmail.com>
References: <cover.1732384228.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 21 Nov 2024 17:02:44 +0000
Subject: [PATCH 33/46] gnu: python-regularizepsf: Update to 1.0.2.

* gnu/packages/astronomy.scm (python-regularizepsf): Update to 1.0.2.
[source]: Swap to PyPI tarball as a fresh release contains tests now.
[phases]: Remove all custom ones.
[propagated-inputs]: Remove python-dill and python-lmfit.
[native-inputs]: Remove python-cython and python-pytest-runner; add
python-setuptools, python-setuptools-scm, and python-wheel.

Change-Id: I1c17bef1a1a21665b633d22282a9df37155e4f29
---
 gnu/packages/astronomy.scm | 36 ++++++++++--------------------------
 1 file changed, 10 insertions(+), 26 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 23c8a85386..0391c339cb 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -4408,44 +4408,28 @@ (define-public python-regions
 (define-public python-regularizepsf
   (package
     (name "python-regularizepsf")
-    (version "0.4.0")
+    (version "1.0.2")
     (source
      (origin
-       (method git-fetch) ; no tests data in the PyPI tarball
-       (uri (git-reference
-             (url "https://github.com/punch-mission/regularizepsf")
-             (commit version)))
-       (file-name (git-file-name name version))
+       (method url-fetch)
+       (uri (pypi-uri "regularizepsf" version))
        (sha256
-        (base32 "0b16lscrzd1lribwis19y6dh6qrgddhcinlc2lbwkzzqqkjdnyzi"))))
+        (base32 "1ial8i9nshhpn3lsgnjqm94dfrzxwz2qgpd8bjzmml1ls0j7sm9v"))))
     (build-system pyproject-build-system)
-    (arguments
-     (list
-      #:phases
-      #~(modify-phases %standard-phases
-         (add-after 'unpack 'relax-requirements
-           (lambda _
-             (substitute* "setup.py"
-               ;; numpy==1.26.4
-               (("==1.26.4") ">=1.23"))))
-          (add-before 'check 'build-extensions
-            (lambda _
-              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+    (native-inputs
+     (list python-pytest
+           python-pytest-mpl
+           python-setuptools
+           python-setuptools-scm
+           python-wheel))
     (propagated-inputs
      (list python-astropy
-           python-dill
            python-h5py
-           python-lmfit
            python-matplotlib
            python-numpy
            python-scikit-image
            python-scipy
            python-sep-pjw))
-    (native-inputs
-     (list python-cython
-           python-pytest
-           python-pytest-mpl
-           python-pytest-runner))
     (home-page "https://github.com/punch-mission/regularizepsf")
     (synopsis "Point spread function modeling and regularization")
     (description
-- 
2.46.0

